### PR TITLE
Prefetch state claims to change O(n) calls to Lotus State to O(1)

### PR DESCRIPTION
**Pull request by RobQuistNL**
In my case (140.000 updates) shaves off 1.5 seconds per call (+ the load on the daemon) -> turns a 39 hour job into seconds

### CODE REVIEW

**Comment by LexLuthr** [comment]
@RobQuistNL Thank you for the PR. Looks good.

File: piecedirectory/doctor.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ import (
 	"github.com/filecoin-project/lotus/chain/types"
 	"github.com/ipfs/go-cid"
 	logging "github.com/ipfs/go-log/v2"
+	verifregtypes "github.com/filecoin-project/go-state-types/builtin/v9/verifreg"
```
Comment (LexLuthr) [inline piecedirectory/doctor.go:20]: Can you please check with v12 verifreg? We are trying to use latest actors as much as possible.

**Review by LexLuthr — COMMENTED** [review]
One nit about the actor version. Rest is good.

File: piecedirectory/doctor.go (hunk 1 above)
Comment (RobQuistNL) [reply piecedirectory/doctor.go:20]: Are you sure that will work? The v9 one is the one used in the return type of the API call

Comment (LexLuthr) [reply piecedirectory/doctor.go:20]: Ahhh! Looks like a Lotus side dependency. Merging.

**Review by LexLuthr — APPROVED** [review]

**Pull request merged**